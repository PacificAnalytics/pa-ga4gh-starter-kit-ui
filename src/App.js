import React from 'react';
import './App.css';
import '@fontsource/roboto';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import { ThemeProvider } from '@material-ui/styles';
import {
    Landing,
    Main
} from './lib/components/pages';
import theme from './lib/styles/theme/theme';

const App = () => {
    return (
        <ThemeProvider theme={theme}>
            <Router>
                <Switch>
                    <Route exact path='/' component={Landing} />
                    <Route path='/**' component={Main} />
                </Switch>
            </Router>
        </ThemeProvider>
    );
}

export default App;
