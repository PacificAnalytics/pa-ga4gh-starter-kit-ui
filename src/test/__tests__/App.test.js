import React from 'react';
import 'babel-polyfill';
import {MemoryRouter} from 'react-router-dom';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-dom'
import App from '../../App';
import userEvent from '@testing-library/user-event';

test('Default route is rendered correctly', () => {
    const history = createMemoryHistory();
    let {container} = render(
        <Router history={history}>
            <App />
        </Router>
    );
    // TODO re-enable tests later 
    // expect(container.firstChild).toMatchSnapshot();
    // expect(screen.getByRole('heading', {level: 1})).toHaveTextContent('Welcome to the GA4GH Starter Kit');
    // userEvent.click(screen.getByRole('button'));
    // expect(screen.getByRole('heading', {level: 2})).toHaveTextContent('Welcome to DRS Starter Kit');
});