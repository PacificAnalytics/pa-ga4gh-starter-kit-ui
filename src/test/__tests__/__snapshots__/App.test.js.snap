// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default route and navigation to /drs are rendered correctly 1`] = `
<main
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    class="MuiGrid-root makeStyles-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
  />
  <div
    class="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
  >
    <div
      class="makeStyles-paper-3"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        GA4GH Starter Kit
      </h1>
      <form
        class="makeStyles-form-5"
        novalidate=""
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-6 MuiButton-containedPrimary MuiButton-fullWidth"
          href="/home"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Enter
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            Copyright Â© Global Alliance for Genomics and Health 
            2021
            .
          </p>
        </div>
      </form>
    </div>
  </div>
</main>
`;
